Add Curve 2023 annual review

aspirer opened the pull request:
related to issue: https://github.com/opencurve/curve/issues/2692

TheFoxAtWork reviewed (commented):
Annual Review Summary
The below is a brief summarization of Curve 2023 Annual Review Evaluation. It contains recommendations the project should strongly consider in order to achieve preparation for Incubation level.

The project scope and goals remain consistent, with community development and growth ongoing. It is worth noting that the bi-weekly meetings repository for the project which is linked throughout the repos appears to have been archived in September with no further record of meetings beyond August.  The project appears to have appropriate governance for its maturity level however the maintainer records do not appear up to date across the repositories and docs (likely also the result of this PR also being open for a period of time). It was not discoverable if this project engages in long term planning. While the project uses tags and milestones for tracking versioned releases and development activities towards that, it was not immediately clear what further improvements are planned, and when they are anticipated to occur or initiate.  Regarding adoption, it is not immediately clear if adoption is picking up beyond the information provided in this PR.

Recommendations:
- Adopt the CNCF code of conduct. the project may retain the code of conduct committee they currently have, however Recommend they list the members of the alias in use for reporting to allow reporters peace of mind when submitting reports.
- Update appropriate repository with meeting notes, information, times, etc. for contributors to discover meetings and review past notes.
- verify and update the maintainer listings for all repos of the project, as appropriate.
- develop a roadmap or other long-term planning document that lays out the desired improvements to Curve as it progresses towards Incubation maturity.
- create an ADOPTERS.md file to track documented adopters of Curve.
- [edit] Continue efforts to identify contributors interested in moving to maintainership.

aspirer commented:
> * Update appropriate repository with meeting notes, information, times, etc. for contributors to discover meetings and review past notes.
* We changed the repo for storing documents, meeting notices, meeting minutes, etc. to the `curve-book` . As you can see, we froze the previous repo `curve-meetup-slides`. The information in the `curve-book` will be automatically published to our project website: https://docs.opencurve.io/category/2023
* We also added a simple migration statement(In Simple Chinese) to the README document of the previous repository.
> * verify and update the maintainer listings for all repos of the project, as appropriate.
* done
> * develop a roadmap or other long-term planning document that lays out the desired improvements to Curve as it progresses towards Incubation maturity.
* https://docs.opencurve.io/Roadmap/roadmap-2023
* the roadmap of next year will be published soon.
> * create an ADOPTERS.md file to track documented adopters of Curve.
* https://github.com/opencurve/community/blob/master/ADOPTERS.md
